pkg/execute.go

```diff
@@ -135,6 +136,25 @@ func Run(cfgPath,
 	return nil
 }
 
+// terraform-exec does not pass through all variables with tf.SetEnv https://github.com/hashicorp/terraform-exec/issues/337
+// so this function combines the existing os.Environ variable list with AWS access & secret key for usage in
+// terraform_remote_state datasources
+func combineEnvVariables(creds TfCreds) map[string]string {
+	ret := make(map[string]string)
+	for _, env := range os.Environ() {
+		split := strings.Split(env, "=")
+		// ignore any TF_ prefixed variables as tfexec will error
+		// https://github.com/hashicorp/terraform-exec/blob/main/tfexec/cmd.go#L22-L39
+		if len(split) > 1 && strings.HasPrefix(split[0], "TF_") {
```

Comment:
Looks like this will only include variables prefixed with TF_ which is opposite of what the comment says?

Reply:
Whoops, forgot the !